### PR TITLE
[POC] Add activatorRef to Secondary, Grouped actions

### LINKED ISSUE #8105: [a11y][Page] Secondary actions should have an ability to receive focus
## Issue summary

There is a focus management issue prevalent across the Admin. Many of the Page secondary actions in the Admin - e.g. Import, Export, Duplicate, etc. - open a modal, and currently there is no easy way to [return focus to the modal activator](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/#keyboard-interaction-7) when the activator is the secondary action of the Page component that doesn't support it.

## Expected behavior

Ability to connect Page secondary actions to the modal or any other overlay to be able to return focus to the activator. 
For example, by adding the ability to provide an `activatorRef` to `<Page />`'s `secondaryActions` to then pass it to the `<Modal />` as an `activator`.

## Actual behavior
No ability to connect `<Page />`'s secondary actions to the overlay they open.

## Steps to reproduce the problem

1. Open any section that has secondary actions that open a modal.
  1.1. Example actions: Export, Import, Duplicate. 
  1.2. Example pages: Discount index, Product index, Product details, Variant details.
1. Use keyboard tab to navigate through the page
2. Activate the action to open a modal
3. Close modal or cancel out of it
4. See the focus returns to the top of the page instead of the modal activator

<details>
      <summary>Product export modal focus</summary>
      <video src="https://user-images.githubusercontent.com/15054990/188015493-77e0a4ff-d05b-4462-8eff-14e8d6a5c6cc.mp4">
</details>

## Relevant discussions and previous attempts to fix it
https://github.com/Shopify/polaris/issues/8058, https://github.com/Shopify/polaris/pull/5258, https://github.com/Shopify/polaris/pull/5495, https://github.com/Shopify/web-configs/pull/349

### PR DESCRIPTION
**Pull request by oksanashopify**
### WHY are these changes introduced?

Resolves https://github.com/Shopify/polaris/issues/8105

TL;DR: There is a focus management issue prevalent across the Admin. Many of the Page secondary actions in the Admin - e.g. Import, Export, Duplicate, etc. - open a modal, and currently there is no easy way to [return focus to the modal activator](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/#keyboard-interaction-7) when the activator is the secondary action of the Page component that doesn't support it.

Here is [related WEB PR](https://github.com/Shopify/web/pull/82434) with implementation of new changes and described test cases

### How to 🎩
Instructions how to setup environment and spin up web and polaris - [Working with Polaris](https://web.docs.shopify.io/docs/guides/working-with-polaris) 

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx [hunk 1]
```diff
@@ -70,6 +73,7 @@ export function MenuGroup({
       accessibilityLabel={accessibilityLabel}
       onClick={handleClick}
       getOffsetWidth={handleOffsetWidth}
+      activatorRef={groupActivatorRef}
```
Comment (oksanashopify) [inline polaris-react/src/components/ActionMenu/components/MenuGroup/MenuGroup.tsx:76]: Allows to return focus to the group of actions 

https://user-images.githubusercontent.com/104942025/226448012-05c7a937-c64c-43e9-8dff-ba21e55ad83a.mov

File: polaris-react/src/components/ActionMenu/components/SecondaryAction/SecondaryAction.tsx [hunk 2]
```diff
@@ -30,9 +32,11 @@ export function SecondaryAction({
...
 
   const buttonMarkup = (
-    <Button onClick={onAction} {...rest}>
-      {children}
-    </Button>
+    <span ref={activatorRef}>
+      <Button onClick={onAction} {...rest}>
+        {children}
+      </Button>
+    </span>
```
Comment (oksanashopify) [inline polaris-react/src/components/ActionMenu/components/SecondaryAction/SecondaryAction.tsx:39]: Allows to return focus to the action within Secondary actions

https://user-images.githubusercontent.com/104942025/226448407-69383eed-d9f3-4e88-9041-d021c6472ced.mov

File: polaris-react/src/components/Button/utils.tsx [hunk 3]
```diff
@@ -30,8 +30,10 @@ export function buttonFrom(
   key?: any,
 ) {
   return (
-    <Button key={key} onClick={onAction} {...action} {...overrides}>
-      {content}
-    </Button>
+    <span ref={action?.activatorRef}>
```
Comment (oksanashopify) [inline polaris-react/src/components/Button/utils.tsx:33]: Allows to return focus to "Archive" and "Delete" product buttons

https://user-images.githubusercontent.com/104942025/226450078-d2b92d1b-283a-404c-9d83-631ce9d130b4.mov
